### PR TITLE
feat: support command

### LINKED ISSUE #458: Add support server command
Add the support server link to a command:

s!support
s!server
s!supportserver

### PR DESCRIPTION
**Pull request by cfanoulis**
I mean, it's what the title says.

Closes #458
## Changelog?
Why, yes
### Additions

 - feat: support command ([4bf59bc](https://github.com/kyranet/Skyra/commit/4bf59bc14607524f849e6ea2ebd46c78aed17a8d))
 - chore: remove unneeded comment ([98ab148](https://github.com/kyranet/Skyra/commits/98ab14894b29c0ab0c60fa5146a7b12aa60efab8))
 - fix: use language builder and adhere to form ([d68d5aa](https://github.com/kyranet/Skyra/commits/d68d5aad86da5dcdff93f9e9796eed3d166ea258))
 - fix: requested changes ([4200e21](https://github.com/kyranet/Skyra/commits/4200e214d6587cba6f6daf456a1c202b13a66ddd))
 - feat: more aliases for support ([bc7fb83](https://github.com/kyranet/Skyra/commits/bc7fb83f5390a60d44053f381d886b5f62c26a77))
 - fix: move language desc keys around ([b98d301](https://github.com/kyranet/Skyra/commits/b98d301e698ee7f7ea3ec1a0583dc8ad57e568c7))
 - feat: "spanish" translation ([95839a7](https://github.com/kyranet/Skyra/commits/95839a70268ba5f3c269cd38be750d7ede501de4))
 - refactor: apply suggestings from code review ([4b0e249](https://github.com/kyranet/Skyra/commits/4b0e249df96cf22a658d2f01a1db9c8278c753ca))

### CODE REVIEW

File: src/commands/System/support.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+			extendedHelp: language => language.tget('COMMAND_SUPPORT_EXTENDED_HELP'),
+			guarded: true,
+			permissionLevel: 0
+		});
+
+	}
+
+	public run(message: KlasaMessage) {
+		const embed = new MessageEmbed()
+		// TODO: LANGUAGE SUPPORT YOU IDIOT
```
Comment (PyroTechniac) [inline src/commands/System/support.ts]: Unneeded comment

**Review by PyroTechniac — CHANGES_REQUESTED** [review]

File: src/commands/System/support.ts (hunk 1 above)
Comment (cfanoulis) [reply src/commands/System/support.ts]: resolved in 98ab148

File: src/commands/System/support.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+		});
+
+	}
+
+	public run(message: KlasaMessage) {
+		const embed = new MessageEmbed()
+			.setTitle(message.language.tget('COMMAND_SUPPORT_EMBED_TITLE', message.author.username))
+			.setDescription(message.language.tget('COMMAND_SUPPORT_EMBED_DESCRIPTION'))
+			.setColor(BrandingColors.Primary);
+		return message.send(embed);
```
Comment (kyranet) [inline src/commands/System/support.ts]: The style in all commands is to inline the embed into the `send` when it doesn't have any conditionals. And we always use `getColor(message)` for the color.

Also, when using embeds, please use `message.sendEmbed()` instead.

```typescript
return message.sendEmbed(new MessageEmbed()
	.setTitle(message.language.tget('COMMAND_SUPPORT_EMBED_TITLE', message.author.username))
	.setDescription(message.language.tget('COMMAND_SUPPORT_EMBED_DESCRIPTION'))
	.setColor(getColor(message)));
```

**Review by kyranet — CHANGES_REQUESTED** [review]

File: src/languages/en-US.ts [hunk 3]
```diff
@@ -1970,6 +1970,10 @@ export default class extends Language {
 		COMMAND_HELP_TITLE: (name, description) => `📃 | ***Help Message*** | __**${name}**__\n${description}\n`,
 		COMMAND_HELP_USAGE: usage => `📝 | ***Command Usage***\n\`${usage}\`\n`,
 		COMMAND_HELP_EXTENDED: extendedHelp => `🔍 | ***Extended Help***\n${extendedHelp}`,
+		COMMAND_SUPPORT_DESCRIPTION: `Show support instructions`,
+		COMMAND_SUPPORT_EXTENDED_HELP: `Literally gives you a link to our support server. What did you really expect this to do :thinking:`,
```
Comment (PyroTechniac) [inline src/languages/en-US.ts]:
```suggestion
		COMMAND_SUPPORT_EXTENDED: `Literally gives you a link to our support server. What did you really expect this to do :thinking:`,
```

This should be renamed to match the consistency of the rest of the files, and probably use the language builder as well

**Review by PyroTechniac — CHANGES_REQUESTED** [review]

File: src/languages/en-US.ts (hunk 3 above)
Comment (cfanoulis) [reply src/languages/en-US.ts]: resolved in d68d5aa

File: src/languages/en-US.ts [hunk 4]
```diff
@@ -1970,6 +1970,12 @@ export default class extends Language {
 		COMMAND_HELP_TITLE: (name, description) => `📃 | ***Help Message*** | __**${name}**__\n${description}\n`,
 		COMMAND_HELP_USAGE: usage => `📝 | ***Command Usage***\n\`${usage}\`\n`,
 		COMMAND_HELP_EXTENDED: extendedHelp => `🔍 | ***Extended Help***\n${extendedHelp}`,
+		COMMAND_SUPPORT_DESCRIPTION: `Show support instructions`,
+		COMMAND_SUPPORT_EXTENDED: builder.display('support', {
+			extendedHelp: `Literally gives you a link to our support server. What did you really expect this to do :thinking:`
+		}),
+		COMMAND_SUPPORT_EMBED_TITLE: username => `Looking for help, ${username}?`,
+		COMMAND_SUPPORT_EMBED_DESCRIPTION: `Then you should probably join [Skyra\'s lounge](https://discord.gg/92mEkYz)! There, you can receive support by the developers and other members of the community!`,
```
Comment (kyranet) [inline src/languages/en-US.ts]: Please port those to `es-ES.ts` so no keys are missing.

Comment (kyranet) [inline src/languages/en-US.ts]:
```suggestion
		COMMAND_SUPPORT_EMBED_DESCRIPTION: `Then you should probably join [Skyra's lounge](https://skyra.pw/join)! There, you can receive support by the developers and other members of the community!`,
```

File: src/languages/en-US.ts [hunk 5]
```diff
@@ -1970,6 +1970,12 @@ export default class extends Language {
 		COMMAND_HELP_TITLE: (name, description) => `📃 | ***Help Message*** | __**${name}**__\n${description}\n`,
 		COMMAND_HELP_USAGE: usage => `📝 | ***Command Usage***\n\`${usage}\`\n`,
 		COMMAND_HELP_EXTENDED: extendedHelp => `🔍 | ***Extended Help***\n${extendedHelp}`,
+		COMMAND_SUPPORT_DESCRIPTION: `Show support instructions`,
+		COMMAND_SUPPORT_EXTENDED: builder.display('support', {
+			extendedHelp: `Literally gives you a link to our support server. What did you really expect this to do :thinking:`
```
Comment (kyranet) [inline src/languages/en-US.ts]: This should have some information about the official server, plus, the message is a little mean, and we do not use `:thinking:` emojis but rather use their unicode.

File: src/languages/en-US.ts (hunk 4 above)
Comment (kyranet) [reply src/languages/en-US.ts]: Also, those lines should move together with the other descriptions.

**Review by kyranet — CHANGES_REQUESTED** [review]

**Comment by cfanoulis** [comment]
please do not merge yet.

**Comment by kyranet** [comment]
We won't while reviews are still not solved, @cfanoulis.

File: src/commands/System/support.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { SkyraCommand } from '../../lib/structures/SkyraCommand';
+import { CommandStore, KlasaMessage } from 'klasa';
+import { MessageEmbed } from 'discord.js';
+import { getColor } from '../../lib/util/util';
+
+export default class extends SkyraCommand {
+
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			aliases: ['supportserver', 'server'],
```
Comment (PyroTechniac) [inline src/commands/System/support.ts]:
```suggestion
			aliases: ['support-server', 'server'],
```
Nit: small thing to add both `support-server` and `supportserver` as aliases

**Review by PyroTechniac — CHANGES_REQUESTED** [review]

File: src/languages/en-US.ts (hunk 5 above)
Comment (favna) [inline src/languages/en-US.ts]:
```suggestion
			extendedHelp: `Gives you a link to the Skyra support server.`
```

File: src/languages/en-US.ts (hunk 4 above)
Comment (favna) [inline src/languages/en-US.ts]:
```suggestion
		COMMAND_SUPPORT_EMBED_DESCRIPTION: `Join [Skyra's lounge](https://skyra.pw/join), where you can receive support from the developers and other members of the community.`,
```

**Review by favna — CHANGES_REQUESTED** [review]
We can do better on the text

File: src/languages/en-US.ts (hunk 4 above)
Comment (cfanoulis) [reply src/languages/en-US.ts]: appplied @kyranet's suggestion on this string

File: src/languages/en-US.ts [hunk 7]
```diff
@@ -602,6 +602,16 @@ export default class extends Language {
 				their marks in a horizontal, vertical, or diagonal row wins the game.`
 		}),
 
+		/**
+		 * ###################
+		 * GENERAL COMMNADS
```
Comment (kyranet) [inline src/languages/en-US.ts]:
```suggestion
		 * GENERAL COMMANDS
```

File: src/languages/en-US.ts [hunk 8]
```diff
@@ -602,6 +602,16 @@ export default class extends Language {
 				their marks in a horizontal, vertical, or diagonal row wins the game.`
 		}),
 
+		/**
+		 * ###################
```
Comment (kyranet) [inline src/languages/en-US.ts]: Always aligned to the title's length:
```suggestion
		 * ################
```

File: src/languages/es-ES.ts [hunk 9]
```diff
@@ -589,6 +589,16 @@ export default class extends Language {
 				their marks in a horizontal, vertical, or diagonal row wins the game.`
 		}),
 
+		/**
+		 * ###################
```
Comment (kyranet) [inline src/languages/es-ES.ts]: Always aligned to the title's length:
```suggestion
		 * ################
```

File: src/languages/es-ES.ts [hunk 10]
```diff
@@ -589,6 +589,16 @@ export default class extends Language {
...
+		 * GENERAL COMMNADS
```
Comment (kyranet) [inline src/languages/es-ES.ts]:
```suggestion
		 * GENERAL COMMANDS
```

File: src/languages/es-ES.ts [hunk 11]
```diff
@@ -1948,6 +1958,8 @@ export default class extends Language {
 		COMMAND_HELP_TITLE: (name, description) => `📃 | ***Mensaje de Ayuda*** | __**${name}**__\n${description}\n`,
 		COMMAND_HELP_USAGE: usage => `📝 | ***Uso del Comando***\n\`${usage}\`\n`,
 		COMMAND_HELP_EXTENDED: extendedHelp => `🔍 | ***Descripción Extendida***\n${extendedHelp}`,
+		COMMAND_SUPPORT_EMBED_TITLE: username => `Nesecita ayuda, ${username}?`,
```
Comment (kyranet) [inline src/languages/es-ES.ts]:
```suggestion
		COMMAND_SUPPORT_EMBED_TITLE: username => `¿Necesita ayuda, ${username}?`,
```

File: src/languages/es-ES.ts [hunk 12]
```diff
@@ -1948,6 +1958,8 @@ export default class extends Language {
...
+		COMMAND_SUPPORT_EMBED_DESCRIPTION: `¡Entonces deberías unirte a [Skyra \ 's lounge] (https://skyra.pw/join)! Allí, puede recibir apoyo de los desarrolladores y otros miembros de la comunidad.`,
```
Comment (kyranet) [inline src/languages/es-ES.ts]:
```suggestion
		COMMAND_SUPPORT_EMBED_DESCRIPTION: `¡Entonces deberías unirte a [Skyra's lounge](https://skyra.pw/join)! Allí, puede recibir ayuda de los desarrolladores y otros miembros de la comunidad.`,
```

File: src/lib/types/Languages.d.ts [hunk 13]
```diff
@@ -169,6 +169,10 @@ export interface LanguageKeys {
 	COMMAND_SKIP_VOTES_VOTED: string;
 	COMMAND_SKIP_VOTES_TOTAL: (amount: number, needed: number) => string;
 	COMMAND_SKIP_SUCCESS: (title: string) => string;
+	COMMAND_SUPPORT_DESCRIPTION: string;
+	COMMAND_SUPPORT_EXTENDED: string;
```
Comment (kyranet) [inline src/lib/types/Languages.d.ts]: Those two should go with the rest.

File: src/languages/en-US.ts [hunk 14]
```diff
@@ -1970,6 +1980,8 @@ export default class extends Language {
 		COMMAND_HELP_TITLE: (name, description) => `📃 | ***Help Message*** | __**${name}**__\n${description}\n`,
 		COMMAND_HELP_USAGE: usage => `📝 | ***Command Usage***\n\`${usage}\`\n`,
 		COMMAND_HELP_EXTENDED: extendedHelp => `🔍 | ***Extended Help***\n${extendedHelp}`,
+		COMMAND_SUPPORT_EMBED_TITLE: username => `Looking for help, ${username}?`,
+		COMMAND_SUPPORT_EMBED_DESCRIPTION: `Then you should probably join [Skyra's lounge](https://skyra.pw/join)! There, you can receive support by the developers and other members of the community!`,
```
Comment (kyranet) [inline src/languages/en-US.ts]:
```suggestion
		COMMAND_SUPPORT_EMBED_DESCRIPTION: `Then you should probably join [Skyra's Lounge](https://skyra.pw/join)! There, you can receive support by the developers and other members of the community!`,
```

File: src/languages/es-ES.ts [hunk 15]
```diff
@@ -589,6 +589,16 @@ export default class extends Language {
...
 		}),
 
+		/**
+		 * ###################
+		 * GENERAL COMMNADS
+		 */
+
+		COMMAND_SUPPORT_DESCRIPTION: `Muestra instrucciones de soporte`,
+		COMMAND_SUPPORT_EXTENDED: builder.display('support', {
+			extendedHelp: `Le brinda un enlace a Skyra's Lounge, su única parada para todos sus problemas e ideas de Skyra`
```
Comment (kyranet) [inline src/languages/es-ES.ts]:
```suggestion
			extendedHelp: `Le brinda un enlace a *Skyra's Lounge*, el lugar indicado para todo lo relacionado conmigo.`
```

**Review by favna — DISMISSED** [review]

File: src/languages/en-US.ts [hunk 16]
```diff
@@ -602,6 +602,16 @@ export default class extends Language {
...
 		}),
 
+		/**
+		 * ###################
+		 * GENERAL COMMNADS
+		 */
+
+		COMMAND_SUPPORT_DESCRIPTION: `Show support instructions`,
+		COMMAND_SUPPORT_EXTENDED: builder.display('support', {
+			extendedHelp: `Gives you a link to the Skyra support server.`
```
Comment (kyranet) [inline src/languages/en-US.ts]:
```suggestion
			extendedHelp: `This command gives you a link to *Skyra's Lounge*, the best place for everything related to me.`
```

**Review by kyranet — CHANGES_REQUESTED** [review]
nits x10

File: src/lib/types/Languages.d.ts (hunk 13 above)
Comment (cfanoulis) [reply src/lib/types/Languages.d.ts]: uh... what do you mean

Comment (kyranet) [reply src/lib/types/Languages.d.ts]: With the rest of the `*_DESCRIPTION` and `*_EXTENDED` strings.

**Review by kyranet — APPROVED** [review]

**Pull request merged**